### PR TITLE
migrating candlepin to pipeline

### PR DESCRIPTION
**Pull request by Lorquas**
This moves the majority of the candlepin jobs to a Jenkinsfile pipeline in their own repo (the same way subscription-manager does).
It renames the 'candlepin' tab to 'Devel'
It adds a 5 minute interval to scan for new PRs

This is required for migrating to the new jenkins instance as ghprb and a few other plugins no longer work.

This will be done in conjunction with a PR in the candlepin repo https://github.com/candlepin/candlepin/pull/2813

### CODE REVIEW

**Comment by Lorquas** [comment]
@cnsnyder  Can you look over the subman parts of this?  (tab name change and poll interval)

File: src/jobs/candlepinPipelineJob.groovy [hunk 1]
```diff
@@ -1,49 +1,54 @@
...
+        discardOldItems {
+            numToKeep(5)
+        }
     }
-    definition {
-        cps {
-            script(readFileFromWorkspace('src/resources/candlepinPipeline.groovy'))
+    triggers {
+        periodicFolderTrigger {
+            interval('5m')
```
Comment (nikosmoum) [inline src/jobs/candlepinPipelineJob.groovy:43]: So this scans every 5 minutes for new PRs and launches a new pipeline for each one?

**Review by nikosmoum — APPROVED** [review]
ACK from me. I'll let @cnsnyder review too

File: src/jobs/candlepinPipelineJob.groovy (hunk 1 above)
Comment (Lorquas) [reply src/jobs/candlepinPipelineJob.groovy:43]: @nikosmoum  Correct.  We can change this to whatever time interval you want, but I think 5 minutes is a good balance of rapid turn around and not constantly using resources just to scan.

Also we can scan for specific branch names (or a regex) and test upon changes to those if you want that.

**Review by cnsnyder — APPROVED** [review]

**Pull request merged**